apps/mail/components/mail/popup-thread-display.tsx

```
@@ -0,0 +1,77 @@
+import { useQueryState } from 'nuqs';
+import { useMemo } from 'react';
+
+export function PopupThreadDisplay() {
+  const [id] = useQueryState('threadId');
+  const { data: emailData, isLoading } = useThread(id ?? null);
+  
+  const allThreadAttachments = useMemo(() => {
+    if (!emailData?.messages) return [];
+    return emailData.messages.reduce<Attachment[]>((acc, message) => {
+      if (message.attachments && message.attachments.length > 0) {
+        return [...acc, ...message.attachments];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)